pkg/controllers/event/serviceaccount.go

```diff
@@ -0,0 +1,59 @@
+package event
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"open-cluster-management.io/managed-serviceaccount/pkg/common"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &serviceAccountEventHandler{}
+
+func NewServiceAccountEventHandler(clusterName string) handler.EventHandler {
+	return serviceAccountEventHandler{
+		clusterName: clusterName,
+	}
+}
+
+type serviceAccountEventHandler struct {
+	clusterName string
+}
+
+func (s serviceAccountEventHandler) Create(event event.CreateEvent, q workqueue.RateLimitingInterface) {
+	serviceAccount, ok := event.Object.(*corev1.ServiceAccount)
+	if ok {
+		s.process(serviceAccount, q)
+	}
+}
+
+func (s serviceAccountEventHandler) Update(event event.UpdateEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (s serviceAccountEventHandler) Delete(event event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	serviceAccount, ok := event.Object.(*corev1.ServiceAccount)
+	if ok {
+		s.process(serviceAccount, q)
+	}
+}
+
+func (s serviceAccountEventHandler) Generic(event event.GenericEvent, q workqueue.RateLimitingInterface) {
+	serviceAccount, ok := event.Object.(*corev1.ServiceAccount)
+	if ok {
+		s.process(serviceAccount, q)
+	}
+}
+
+func (s serviceAccountEventHandler) process(serviceAccount *corev1.ServiceAccount, q workqueue.RateLimitingInterface) {
+	if serviceAccount.Labels[common.LabelKeyIsManagedServiceAccount] != "true" {
```

Comment:
i think we need a nil check on the labels field otherwise will be a nil panic

Reply:
Labels is of type `Labels map[[string]` which is not nil able... but I suppose we should a nil check for `serviceAccount`

Reply:
oh u r right, dang i was doing the nil check on the map all the yrs..